utils/localStorage.js

```diff
@@ -48,21 +48,33 @@
         }
     }
 
-    function getQueriesFromLocalStorage(storageName, term) {
+    function getQueriesFromLocalStorage(storageName, term, target) {
         var queries = getSuggestionQueries(storageName);
         if (queries !== null) {
-            term = escapeSpecialChars(term);
             var matchedQueries = queries.map(function (query) {
-                query = JSON.stringify(query);
-                var regex = new RegExp(term);
-                var match = regex.exec(query);
-                if (match !== null) {
-                    if (term.length) {
-                        query = query.replace(match[0], '<b>'.concat(match[0], '</b>'));
+                var match
+                var regex
+                if (target !== null && query[target]) {
+                    regex = new RegExp('^'+escapeSpecialChars(term[target]));
+                    match = regex.exec(query[target]);
+                    if (match !== null) {
+                        query[target] = query[target].replace(match[0], '<b>'.concat(match[0], '</b>'))
+                        query.isQueryHistory = true;
+                        return query
+                    }
+                } else {
+                    term = escapeSpecialChars(term);
+                    query = JSON.stringify(query);
+                    regex = new RegExp(term)
+                    match = regex.exec(query)
+                    if (match !== null) {
+                        if (term.length) {
+                            query = query.replace(match[0], '<b>'.concat(match[0], '</b>'));
+                        }
+                        query = JSON.parse(query);
+                        query.isQueryHistory = true;
+                        return query;
```

Comment:
Is this necessary? I think to return all the values when we don't have a prefix. But if we have some characters we must check only the prefix. What do you think?

Reply:
Indeed, in Jus, the expected is that it always falls in the first case. I will remove this condition! :+1: